### PR TITLE
UIREQ-978: Prepare request page for future changes

### PR DESCRIPTION
**Pull request by artem-blazhko**
## Purpose
Prepare request page for future changes.

## Approach
- Change the order of main elements on the page
- Add request type select box 
- Add validation to request type select box
- Delete unnecessary static content
- Delete unnecessary big tests
- Modify big tests
- Cover new functionality by Jest/RTL tests

## Refs
[UIREQ-978](https://issues.folio.org/browse/UIREQ-978)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@
 * Cover InstanceInformation by jest/RTL tests. Refs UIREQ-950.
 * Fix inconsistency in RTL/Jest tests. Refs UIREQ-979.
 * Remove redundant ariaLabel prop. Refs UIREQ-972.
+* Request page preparation. Refs UIREQ-978.
```
Comment (Dmitriy-Litvinenko) [inline CHANGELOG.md:41]: Could you please change to  "Preparation for Enable Request Policy to Determine Allowed Pickup Service Points. Refs UIREQ-978."?

File: src/constants.js [hunk 2]
```diff
@@ -121,6 +121,18 @@ export const itemStatusesTranslations = {
   [itemStatuses.RESTRICTED]: 'ui-requests.item.status.restricted',
 };
 
+export const requestTypeErrors = {
```
Comment (Dmitriy-Litvinenko) [inline src/constants.js:124]: Could you please rename from requestTypeErrors to REQUEST_TYPE_ERRORS?

File: src/constants.js [hunk 3]
```diff
@@ -121,6 +121,18 @@ export const itemStatusesTranslations = {
...
+  TITLE_LEVEL_ERROR: 'title',
+  ITEM_LEVEL_ERROR: 'item',
+};
+
+export const requestTypeErrorTranslations = {
```
Comment (Dmitriy-Litvinenko) [inline src/constants.js:129]: Could you please rename from requestTypeErrorTranslations to REQUEST_TYPE_ERROR_TRANSLATION_KEYS?

File: src/routes/RequestsRoute.test.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ import {
 } from '../utils';
 import {
   REQUEST_LEVEL_TYPES,
-  createModes,
+  createModes, DEFAULT_REQUEST_TYPE_VALUE,
```
Comment (Dmitriy-Litvinenko) [inline src/routes/RequestsRoute.test.js:27]: Could you please move `DEFAULT_REQUEST_TYPE_VALUE,` to new line?

File: CHANGELOG.md (hunk 1 above)
Comment (artem-blazhko) [reply CHANGELOG.md:41]: fixed

File: src/components/RequestInformation/RequestInformation.js [hunk 5]
```diff
@@ -20,29 +21,48 @@ import {
...
   REQUEST_FORM_FIELD_NAMES,
+  requestTypeErrorTranslations,
+  requestTypeErrors,
 } from '../../constants';
 import PositionLink from '../../PositionLink';
 import { isFormEditing } from '../../utils';
 
 const DATE_FORMAT = 'YYYY-MM-DD';
 
+export const getNoRequestTypeErrorMessageId = (isTitleLevelRequest) => (isTitleLevelRequest ?
```
[the remark this answers is not in the document]
Comment (artem-blazhko) [reply src/components/RequestInformation/RequestInformation.js:32]: done

File: src/constants.js (hunk 2 above)
Comment (artem-blazhko) [reply src/constants.js:124]: done

File: src/constants.js (hunk 3 above)
Comment (artem-blazhko) [reply src/constants.js:129]: done

File: src/routes/RequestsRoute.test.js (hunk 4 above)
Comment (artem-blazhko) [reply src/routes/RequestsRoute.test.js:27]: Done

**Comment by Dmitriy-Litvinenko** [comment]
Hello @folio-org/fe-tl-reviewers Could you please review?

File: src/components/RequestInformation/RequestInformation.js [hunk 6]
```diff
@@ -58,54 +80,54 @@ const RequestInformation = ({
...
+                            id={id}
+                            key={id}
+                          >
+                            {translatedLabel => (
+                              <option value={value}>
+                                {translatedLabel}
+                              </option>
+                            )}
+                          </FormattedMessage>
+                        ))}
```
Comment (mariia-aloshyna) [inline src/components/RequestInformation/RequestInformation.js:126]: Why don't you use `dataOptions` prop to render options?

File: src/components/RequestInformation/RequestInformation.test.js [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+  screen,
+  fireEvent,
+  cleanup,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+
+import { Field } from 'react-final-form';
+
+import { Select } from '@folio/stripes-components';
```
Comment (mariia-aloshyna) [inline src/components/RequestInformation/RequestInformation.test.js:13]:
```suggestion
import { Select } from '@folio/stripes/components';
```

**Comment by artem-blazhko** [comment]
@mariia-aloshyna could you take a look one more time?

File: src/components/RequestInformation/RequestInformation.js (hunk 6 above)
Comment (mariia-aloshyna) [reply src/components/RequestInformation/RequestInformation.js:126]: it will reduce your code by almost 20 lines :)

Comment (artem-blazhko) [reply src/components/RequestInformation/RequestInformation.js:126]: @mariia-aloshyna Some of the options for `Select` component are generated dynamically.
If I create a separate `dataOptions` array we will not reduce my code.

cc @Dmitriy-Litvinenko

Comment (artem-blazhko) [reply src/components/RequestInformation/RequestInformation.js:126]: Furthermore, I see that our code base contains both approaches.

**Review by mariia-aloshyna — APPROVED** [review]

File: src/components/RequestInformation/RequestInformation.js (hunk 6 above)
Comment (mariia-aloshyna) [reply src/components/RequestInformation/RequestInformation.js:126]: I approved this PR, but still, it doesn't look like the additional 20 lines of code that are already implemented by the Select component.
`dataOptions={requestTypeOptions.map(({ id, value }) => ({ label: formatMessage({ id }), value }))}`

**Pull request merged**